File: src/main/java/com/crui/patterns/examples/turnos_medicos/after/Paciente.java
package com.crui.patterns.examples.turnos_medicos.after;

public class Paciente {
  String nombre;
  String obraSocial;
  String email;

  public Paciente(String nombre, String obraSocial, String email) {
    this.nombre = nombre;
    this.obraSocial = obraSocial;
    this.email = email;
  }

  public void avisarCambioDeFechayHora(Turno turno) {
    System.out.println(
        "Mail para "
            + email
            + ": El turno con "
            + turno.getDoctor()
            + " se ha cambiado a "
            + turno.getFechaYHora());
  }

  @Override
  public String toString() {
    return nombre + " (" + obraSocial + ")";
  }
}

File: src/main/java/com/crui/patterns/examples/turnos_medicos/after/Paciente.py
from Turno import Turno

class Paciente:
    def __init__(self, nombre: str, obraSocial: str, email: str):
        self.nombre = nombre
        self.obraSocial = obraSocial
        self.email = email

    def avisarCambioDeFechayHora(self, turno: Turno) -> None:
        print(
            "Mail para "
            + self.email
            + ": El turno con "
            + turno.doctor
            + " se ha cambiado a "
            + turno.fechaYHora
        )

    def __str__(self) -> str:
        return self.nombre + " (" + self.obraSocial + ")"